userspace/libscap/scap_savefile.c

```diff
@@ -50,17 +50,34 @@ int scap_dump_write(scap_dumper_t *d, void* buf, unsigned len)
 	}
 	else
 	{
-		if(d->m_targetbufcurpos + len < d->m_targetbufend)
+		if(d->m_targetbufcurpos + len >= d->m_targetbufend)
 		{
-			memcpy(d->m_targetbufcurpos, buf, len);
+			if(d->m_type == DT_MEM)
+			{
+				return -1;
+			}
 
-			d->m_targetbufcurpos += len;
-			return len;
-		}
-		else
-		{
-			return -1;
+			// DT_MANAGED_BUF, try to increase the size
+			size_t targetbufsize = PPM_DUMPER_MANAGED_BUF_RESIZE_FACTOR * (d->m_targetbufend - d->m_targetbuf);
+
+			uint8_t *targetbuf = (uint8_t *)realloc(

```

Comment:
Here we can see a rare `realloc` usage without the "reassign pointer thus losing it in case of fail" bug :rofl: